### ISSUE TITLE
Add support for notification grouping

### ISSUE THREAD

**Issue by mammuth** [home-assistant/home-assistant-android#148]
I understand that the current implementation of notifications is only a first draft. 

I just want to suggest for discussion / later implementation that we should implement [notification groups](https://developer.android.com/guide/topics/ui/notifiers/notifications#bundle) as seen in the following example.

![image](https://user-images.githubusercontent.com/3121306/70792177-f15f2380-1d98-11ea-920b-3be1f0211314.png)

**Comment by dshokouhi** [comment]
i think grouping would be useful once we have channels setup so we can have each notification channel grouped separately.

**Comment by mammuth** [comment]
Good idea, automatic grouping of channels (#107) might make sense :+1:

**Comment by andriej** [comment]
Isn't the tagging support needed from html5 to "tag" also? So one notification overwrite older one when it comes fo one device? 
I don't know channels but you probably refer to iOS app which I haven't seen.

**Comment by robbiet480** [comment]
`tag` will replace any existing notifications with the same `tag`. This is more akin to threading on iOS. In that vein, I suggest that when this gets implemented we use the same key on Android as we do iOS: `thread-id`.

**Comment by dshokouhi** [comment]
I think `tag` and grouping are 2 different things here.  Tag will replace and remove notification....grouping will behave like the screenshots.  I think that in theory you can use tag and group together.  Now I wonder how the those 2 things will tie into notification channels once we support that?

**Comment by robbiet480** [comment]
Correct, they can be used together. My understanding is that groups and channels don't have bearing on each other.
